### ISSUE TITLE
incorrect error message for duplicate Send impl 

### ISSUE THREAD

**Issue by ultrasaurus** [rust-lang/rust#69938]
For a specific case, the compiler reports an incorrect error message when there is a duplicate error 

I tried this code:

```rust
pub struct Foo(*const ());

unsafe impl Send for Foo {}
unsafe impl Send for Foo {}

impl Foo {
    fn bar(&self) {}
}

fn assert_send<T: Send>(_: T) {}

fn main() {
    let foo = Foo(&());
    assert_send(async move {
         foo.bar();
    })
}
```

it reports this error:

```
  |
10 | fn assert_send<T: Send>(_: T) {}
   |    -----------    ---- required by this bound in `assert_send`
...
13 |     let foo = Foo(&());
   |         --- consider giving `foo` a type
14 |     assert_send(async move {
   |     ^^^^^^^^^^^ cannot infer type for struct `Foo`
   |
   = note: cannot resolve `Foo: std::marker::Send`
   = note: required because it appears within the type `[static generator@src/main.rs:14:28: 16:6 foo:Foo {}]`
   = note: required because it appears within the type `std::future::GenFuture<[static generator@src/main.rs:14:28: 16:6 foo:Foo {}]>`
   = note: required because it appears within the type `impl std::future::Future
```

If I comment out the line that calls foo.bar() I get the expected error:

```
 |
3 | unsafe impl Send for Foo {}
  | ------------------------ first implementation here
4 | unsafe impl Send for Foo {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Foo`
```

Here's a [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=722dc1babdd7e50693b5010882d905c2)

### Meta

Reproduced in 1.14.1, It also happens in nightly

originally found in
`rustc --version --verbose`:
```
rustc --version --verbose
rustc 1.41.0 (5e1a79984 2020-01-27)
binary: rustc
commit-hash: 5e1a799842ba6ed4a57e91f7ab9435947482f7d8
commit-date: 2020-01-27
host: x86_64-apple-darwin
release: 1.41.0
LLVM version: 9.0

```

**Comment by ultrasaurus** [comment]
thx @sgrif for helping to create an isolated example!

**Comment by tmandry** [comment]
Seems related to #67651, and may be fixed by #68884.

**Comment by csmoe** [comment]
Fixed in latest nightly
```txt
error[E0119]: conflicting implementations of trait `std::marker::Send` for type `Foo`:
 --> src/main.rs:4:1
  |
3 | unsafe impl Send for Foo {}
  | ------------------------ first implementation here
4 | unsafe impl Send for Foo {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Foo`
```
